deebot_client/commands/json/map.py

```diff
@@ -55,48 +65,15 @@ def _handle_response(
             return CommandResult(
                 result.state,
                 result.args,
-                [GetMapSet(result.args["map_id"], entry) for entry in MapSetType],
+                # an if check for getmapset, as newer bot use different api version
+                [GetMapSetV2(result.args["map_id"], entry) for entry in MapSetType]
+                if self.get_map_set_version == 2
+                else [GetMapSet(result.args["map_id"], entry) for entry in MapSetType],
```

Comment:
```suggestion
                [self._map_set_command(result.args["map_id"], entry) for entry in MapSetType],
```

I did not test this suggested changes, but imo they should work

Reply:
updated to use the match-case option